android/src/main/java/com/onegini/mobile/sdk/reactnative/OneginiSDK.kt

```diff
@@ -13,24 +13,22 @@ import com.onegini.mobile.sdk.reactnative.handlers.mobileauthotp.MobileAuthOtpRe
 import com.onegini.mobile.sdk.reactnative.handlers.pins.CreatePinRequestHandler
 import com.onegini.mobile.sdk.reactnative.handlers.pins.PinAuthenticationRequestHandler
 import com.onegini.mobile.sdk.reactnative.model.rn.OneginiReactNativeConfig
-
-class OneginiSDK(private val reactApplicationContext: ReactApplicationContext) {
-
-    lateinit var registrationRequestHandler: RegistrationRequestHandler
-        private set
-    lateinit var pinAuthenticationRequestHandler: PinAuthenticationRequestHandler
-        private set
-    lateinit var createPinRequestHandler: CreatePinRequestHandler
-        private set
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class OneginiSDK @Inject constructor(
+    private val reactApplicationContext: ReactApplicationContext,
```

Comment:
application context can be injected directly here:
`private val applicationContext: Context`